SConstruct

```diff
@@ -770,24 +876,50 @@ for variant_path in variant_paths:
     for cb in after_sconsopts_callbacks:
         cb()
 
-    # Update env for new variables added by the SConsopts.
-    sticky_vars.Update(env)
-
-    Help('''
-Build variables for {dir}:
-{help}
-'''.format(dir=variant_dir, help=sticky_vars.GenerateHelpText(env)),
-         append=True)
-
-    # If the old vars file exists, delete it to avoid confusion/stale values.
-    if isfile(build_root_vars):
-        warning(f'Deleting old variant variables file "{build_root_vars}"')
-        remove(build_root_vars)
-    # Save sticky variables back to the gem5.build variant variables file.
-    sticky_vars.Save(gem5_build_vars, env)
-
-    # Pull all the sticky variables into the CONF dict.
-    env['CONF'].update({key: env[key] for key in sticky_vars.keys()})
+    # Handle any requested kconfig action, then exit.
+    if kconfig_action:
+        if kconfig_action == 'defconfig':
+            if len(kconfig_args) != 1:
+                error('Usage: scons defconfig <build dir> <defconfig file>')
+            defconfig_path = makePathAbsolute(kconfig_args[0])
+            kconfig.defconfig(env, kconfig_file.abspath,
+                    defconfig_path, config_file.abspath)
+        elif kconfig_action == 'guiconfig':
+            kconfig.guiconfig(env, kconfig_file.abspath, config_file.abspath,
+                    variant_path)
+        elif kconfig_action == 'listnewconfig':
+            kconfig.listnewconfig(env, kconfig_file.abspath,
+                    config_file.abspath)
+        elif kconfig_action == 'menuconfig':
+            kconfig.menuconfig(env, kconfig_file.abspath, config_file.abspath,
+                    variant_path)
+        elif kconfig_action == 'oldconfig':
+            kconfig.oldconfig(env, kconfig_file.abspath, config_file.abspath)
+        elif kconfig_action == 'olddefconfig':
+            kconfig.olddefconfig(env, kconfig_file.abspath,
+                    config_file.abspath)
+        elif kconfig_action == 'savedefconfig':
+            if len(kconfig_args) != 1:
+                error('Usage: scons defconfig <build dir> <defconfig file>')
+            defconfig_path = makePathAbsolute(kconfig_args[0])
+            kconfig.savedefconfig(env, kconfig_file.abspath,
+                    config_file.abspath, defconfig_path)
+        elif kconfig_action == 'setconfig':
+            kconfig.setconfig(env, kconfig_file.abspath, config_file.abspath,
+                    ARGUMENTS)
+        else:
+            error(f'Unrecognized kconfig action {kconfig_action}')
```

Comment:
I don't think it's possible to hit this line of code because `kconfig_action` will be false if it's not one of the above. I guess there's nothing to change here, but it's a bit strange.